tests/virt_autotest/reset_partition.pm

```diff
@@ -0,0 +1,118 @@
+use strict;
+use warnings;
+use testapi;
+use virt_utils;
+use ipmi_backend_utils;
+use Utils::Architectures;
+use utils qw(zypper_call);
+use version_utils 'is_sle';
+use virt_autotest::utils qw(is_xen_host);
+
+sub reset_partition {
+    my $self       = shift;
+    my $new_disk   = "";
+    my $virt_point = "/var/lib/libvirt/";
+
+    # Install requested pkgs
+    my @pkgs = qw/xfsprogs coreutils util-linux/;
+    push push(@pkgss, 'btrfsprogs') if is_sle('=12-SP5');
+    zypper_call "in @pkgs";
+
+    # Get some debug info about hard disk topology
+    record_info("Report file system disk space", script_output("df"));
+    record_info("Find mounted filesystem",       script_output("findmnt"));
+    record_info("Check with /var/lib/libvirt",   script_output("ls -al /var/lib/libvirt"));
+    script_run 'df --output=avail /var/lib/libvirt';
+
+    my ($DEPLOY_DISK_NAME, $AVAILABLE_DISK_SIZE) = $self->get_active_disk_and_available_space();
+    # Disk which /var/lib/libvirt resides on
+    record_info('Detect Deployed DISK NAME for /var/lib/libvirt:', $DEPLOY_DISK_NAME);
+    # Ensure available disk space under /var/lib/libvirt
+    record_info('Detect Available DISK SIZE under /var/lib/libvirt:', $AVAILABLE_DISK_SIZE . 'GiB');
+    # Define the expected disk space for /var/lib/libvirt both on KVM or XEN
+    my $expected_libvirt_size = (is_xen_host) ? '60' : '40';
+
+    #    if ($AVAILABLE_DISK_SIZE lt $expected_libvirt_size) {
+
+    my @disks = qa/sda sdb sdc/;
+    for my $disks (@disks) {
+        assert_script_run('file' . $disks);
+    }
+
+    if ($DEPLOY_DISK_NAME eq 'sda') {
+        $new_disk = "/dev/sdb";
+    } else {
+        $new_disk = "/dev/sda";
+    }
+
+    # Assign a New FULL Disk for /var/lib/libvirt
+    record_info('Assign a New Disk for /var/lib/libvirt:', $new_disk);
+
+    # Wipe Hard Disk Clean via dd for assigned a new full disk
+    assert_script_run("dd if=/dev/zero of=$new_disk count=1M", timeout => 1500, fail_message => "Failed to wipe hard disk clean on $new_disk");
+    save_screenshot;
+
+    # Create New Disk Partition
+    assert_script_run 'echo -e "g\nn\n\n\n\nt\n8e\np\nw" | fdisk ' . $new_disk;
+
+    # Format New Disk Partition
+    $new_disk .= 1;
+    assert_script_run "mkfs.xfs -f $new_disk";
+    save_screenshot;
+
+    # Umount existed /var/lib/libvirt mounted point
+    assert_script_run "umount -f $virt_point";
+    save_screenshot;
+
+    # Wipe existed /var/lib/libvirt mounted point from /etc/fstab
+    script_run "sed -i '/\\/var\\/lib\\/libvirt/d' /etc/fstab";
+
+    # Add a new /var/lib/libvirt mounted point to /etc/fstab
+    assert_script_run "echo '$new_disk $virt_point xfs defaults 0 0' >> /etc/fstab";
+
+    # Mount the new /var/lib/libvirt mounted point
+    assert_script_run "mount -a";
+    save_screenshot;
+    my @dirs = qw/boot dnsmasq filesystems images network qemu swtpm/;
+    push @dirs, qw(libxl lxc) if get_var("XEN");
+    for my $dirs (@dirs) {
```

Comment:
for => foreach ?  $dir rather than $dirs is more readable here.

Reply:
sure, had updated to used with `foreach $dir (@dirs)` way,